src/main/java/org/folio/services/item/ItemService.java

```diff
@@ -109,6 +110,16 @@ public ItemService(Context vertxContext, Map<String, String> okapiHeaders) {
     return itemRepository.getById(itemId)
       .compose(CommonValidators::refuseIfNotFound)
       .compose(oldItem -> refuseWhenHridChanged(oldItem, newItem))
+      .compose(oldItem -> {
+        var oldDestServicePointId = oldItem.getInTransitDestinationServicePointId();
+        if (newItem.getStatus().getName() == Status.Name.IN_TRANSIT &&
+          newItem.getInTransitDestinationServicePointId() == null) {
+
+          newItem.setInTransitDestinationServicePointId(oldDestServicePointId);
+        }
+
+        return succeededFuture(oldItem);
```

Comment:
Why are we trying to preserve this on the server rather than relying on the client to provide a representation that includes it?